src/DIPS.Xamarin.UI/Controls/TrendGraph/TrendGraph.xaml.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections;
+using Xamarin.Forms;
+using DIPS.Xamarin.UI.Extensions;
+using System.Collections.Specialized;
+
+namespace DIPS.Xamarin.UI.Controls.TrendGraph
+{
+    public partial class TrendGraph : ContentView
+    {
+        private VisualElement? m_previousParent;
+        public TrendGraph()
+        {
+            HorizontalOptions = LayoutOptions.FillAndExpand;
+            VerticalOptions = LayoutOptions.FillAndExpand;
+            SizeChanged += (s, e) => StartRedraw(this, EventArgs.Empty);
+            InitializeComponent();
+        }
+
+        protected override void OnParentSet()
+        {
+            var parentSize = ((VisualElement)Parent).Width;
+            if (m_previousParent != null)
+            {
+                m_previousParent.SizeChanged -= StartRedraw;
+                m_previousParent = null;
+            }
+
+            if (Parent is VisualElement element)
+            {
+                element.SizeChanged += StartRedraw;
+                m_previousParent = element;
+                StartRedraw(this, EventArgs.Empty);
+            }
+
+            base.OnParentSet();
+        }
+
+        private void StartRedraw(object _, EventArgs __)
+        {
+            Redraw();
+        }
+
+        private void Redraw()
+        {
+            graphContainer.Children.Clear();
+            if (MinValue >= MaxValue ||
+                ItemsSource == null ||
+                ItemsSource.Count == 0 ||
+                !IsVisible ||
+                GraphBackgroundColor == GraphColor ||
+                Width < 1)
+            {
+                return;
+            }
+
+            if (ItemsSource is INotifyCollectionChanged collectionChanged)
+            {
+                collectionChanged.CollectionChanged -= CollectionChanged_CollectionChanged;
+                collectionChanged.CollectionChanged += CollectionChanged_CollectionChanged;
+            }
+
+            var amount = ItemsSource.Count;
+            var totalWidth = Width;
+            var height = Height;
+            var margin = GraphMargin;
+            var widthPerItem = totalWidth / (amount + (amount - 1) * margin);
+            margin = widthPerItem * GraphMargin;
+            var x = 0.0;
+            foreach (var item in ItemsSource)
+            {
+                var itemx = x;
```

Comment:
Do we need a local variable for this?

Reply:
Yes, because it's used in the func and updated afterwards. And when the func is run they get the wrong value .